lib/remote/apilistener-configsync.cpp

```diff
@@ -393,6 +395,51 @@ void ApiListener::UpdateConfigObject(const ConfigObject::Ptr& object, const Mess
 	}
 }
 
+/**
+ * Syncs the specified object and its direct and indirect parents to the provided client
+ * in topological order of their dependency graph recursively.
+ *
+ * Objects that the client does not have access to are skipped without going through their dependency graph.
+ *
+ * Please do not use this method to forward remote generated cluster updates; it should only be used to
+ * send local updates to that specific non-nullptr client.
+ *
+ * @param object The config object you want to sync.
+ * @param client The JsonRpc client you send the update to.
+ * @param syncedObjects Used to cache the already synced objects.
+ */
+void ApiListener::UpdateConfigObjectWithParents(const ConfigObject::Ptr& object, const JsonRpcConnection::Ptr& client,
+	std::unordered_set<ConfigObject*>& syncedObjects)
+{
+	if (syncedObjects.find(object.get()) != syncedObjects.end()) {
+		return;
+	}
+
+	VERIFY(client);
+
+	Endpoint::Ptr endpoint = client->GetEndpoint();
+	ASSERT(endpoint);
```

Comment:
Huh? VERIFY here, but ASSERT there?

Reply:
I don't know 🤷! Assert is already used within some methods there and I don't know why I chose to use `VERIFY`.

Reply:
Then you should make an informed decision:

https://github.com/Icinga/icinga2/blob/e50eb52291568056686c65ff836b5593715df515/lib/base/debug.hpp#L8-L14

Btw. I didn't even say one of them or even a mix is wrong. I just wondered about this particular mix.

Also, while on it: Don't you VERIFY/ASSERT (or even GetEndpoint etc.) the same once for every object?